### ISSUE TITLE
A lot of warnings about misaligned address in stb_vorbis.c

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#31776]
**Godot version:**
3.2 master

**OS/device including version:**
Ubuntu 19.04

**Issue description:**
When I compile  Godot with sanitizers support e.g.
```
scons p=x11 -j6 use_ubsan=yes use_lsan=yes use_asan=yes
```
then after importing(first time) project which contains ogg file, a lot of this errors shows:
```
thirdparty/misc/stb_vorbis.c:4181:37: runtime error: member access within misaligned address 0x7ffaf9e17064 for type 'struct Mapping', which requires 8 byte alignment
0x7ffaf9e17064: note: pointer points here
  09 09 00 00 01 00 00 00  00 00 00 00 c4 70 e1 f9  fa 7f 00 00 01 00 00 00  00 00 00 00 00 00 00 00
              ^ 
thirdparty/misc/stb_vorbis.c:4182:22: runtime error: member access within misaligned address 0x7ffaf9e2128c for type 'struct stb_vorbis', which requires 8 byte alignment
0x7ffaf9e2128c: note: pointer points here
  fc 01 fc 03 44 ac 00 00  02 00 00 00 d4 51 03 00  00 10 00 00 71 02 00 00  00 00 00 00 00 00 00 00
              ^
thirdparty/misc/stb_vorbis.c:4511:83: runtime error: left shift of 128 by 24 places cannot be represented in type 'int'
thirdparty/misc/stb_vorbis.c:2097:62: runtime error: load of misaligned address 0x7ffaf9e1691c for type 'uint8 *', which requires 8 byte alignment
0x7ffaf9e1691c: note: pointer points here
  fa 7f 00 00 e0 6a e1 f9  fa 7f 00 00 e4 6a e1 f9  fa 7f 00 00 e8 6a e1 f9  fa 7f 00 00 ec 6a e1 f9
              ^ 
```

**Comment by akien-mga** [comment]
Relevant comment from the upstream thread: https://github.com/nothings/stb/issues/799#issuecomment-526289125

> Everything defaults to being allocated by malloc. Most likely, either Godot engine overrides the allocator (and isn't staying on 8-byte multiples), or the compiler is being told to pack structures instead of respecting natural alignment. (I mean, maybe there's something in stb_vorbis I'm missing/forgetting, but I don't see anything obvious when looking at it.)

**Comment by pwaller** [comment]
I believe I found the cause upstream and submitted a possible fix: https://github.com/nothings/stb/pull/860

**Comment by qarmin** [comment]
It is fixed in 1.19, but needs to be updated in Godot.

**Comment by akien-mga** [comment]
Fixed by #38361.
